ui/src/app/guards/access-tasks-guard.service.ts

```diff
@@ -25,8 +25,11 @@ export class AccessTasksGuard implements CanActivate, CanActivateChild {
     }
     return this.serverConfigService.getConfig().flatMap((config) => {
       return this.profileStorageService.profile$.flatMap((profile) => {
-        if (profile.eraCommonsCompletionTime === null
-            && config.enforceRegistered && environment.enableComplianceLockout) {
```

Comment:
The logic here is starting to scare me a little... for example, do we have a clear understanding of the difference between config.enforceRegistered and environment.enableComplianceLockout? AFAICT, they're only ever checked in tandem on the frontend. And it's strange that "complianceLockout" boolean is controlling eraCommons logic, when eraCommons has nothing to do with compliance training. I worry we're at risk of confusing ourselves into oblivion here if we don't proactively clean this up.

Just brainstorming here, I can think of a couple options:

1) Use dataAccessLevel enum to drive all frontend logic.

Since we're already implementing all of this logic in the backend (in UserService.updateDataAccessLevel), we could just use the output of that method, which sets the dataAccessLevel enum, to drive our guard logic here. For example:

if (profile.dataAccessLevel == DataAccessLevel.UNREGISTERED) {
  // Re-route to homepage, where access tasks would be expected to be shown.
}

2. Use backend config flags to drive all access module logic.

Since we're already propagating the backend config into the frontend via an API method, we could drive all of this lockout from purely the config flags, where we're already set up with per-module flags and a global enforceRegistered flag. (But note that part of our plan is to eventually remove enforceRegistered, since all access control should be mediated by access modules. That's for later, though.) For example:

if (profile.eraCommonsCompletionTime === null && profile.eraCommonsBypassTime === null &&
    config.access.enableEraCommons && config.firecloud.enforceRegistered) {
  // Navigate to homepage where access tasks should be shown.
}


Separate from the above consideration, we also have an issue where much of the access tasks logic is duplicated both here and in views/homepage/component.tsx L361. It would be awesome if we could centralize the logic somehow, maybe by pulling it out into a utility method that takes in the profile & config and decides if the user should be blocked, and which modules should be shown to the user. So something like:

// access-tasks-guard.service.ts
if (!accessUtils.shouldAccessWorkbench(profile, config)) {
  // Navigate to homepage where access tasks should be shown.
}

// homepage/component.tsx
if (!accessUtils.shouldAccessWorkbench(profile, config)) {
  this.setState({
    accessTasksRemaining: true
    accessTasksLoaded: true
  });
}


Thoughts? I'm just thinking aloud here; maybe there are good reasons not to move in this direction, now or ever.